ocs_ci/deployment/fusion_aas.py

```diff
@@ -0,0 +1,157 @@
+        )
+
+        # generate kubeconfig and kubeadmin-password files
+        if config.ENV_DATA["ms_env_type"] == "staging":
+            ocm.get_kubeconfig(self.cluster_name, kubeconfig_path)
+            ocm.get_kubeadmin_password(self.cluster_name, password_path)
+        if config.ENV_DATA["ms_env_type"] == "production":
+            if config.ENV_DATA.get("appliance_mode"):
+                logger.info(
+                    "creating admin account for cluster in production environment with "
+                    "appliance mode deployment is not supported"
+                )
+                return
+            else:
+                rosa_prod_cluster = ROSAProdEnvCluster(self.cluster_name)
+                rosa_prod_cluster.create_admin_and_login()
+                rosa_prod_cluster.generate_kubeconfig_file(skip_tls_verify=True)
+                rosa_prod_cluster.generate_kubeadmin_password_file()
+
+        self.test_cluster()
+
+
+class FUSIONAAS(rosa_deployment.ROSA):
+    """
+    Deployment class for Fusion aaS.
+    """
+
+    OCPDeployment = FUSIONAASOCP
+
+    def __init__(self):
+        self.name = self.__class__.__name__
+        super(FUSIONAAS, self).__init__()
+        ocm.download_ocm_cli()
+        rosa.download_rosa_cli()
+        self.aws = AWSUtil(self.region)
+
+    def deploy_ocp(self, log_cli_level="DEBUG"):
+        """
+        Deployment specific to OCP cluster on a cloud platform.
+
+        Args:
+            log_cli_level (str): openshift installer's log level
+                (default: "DEBUG")
+
+        """
+        ocm.login()
+        super(FUSIONAAS, self).deploy_ocp(log_cli_level)
+        self.host_network_update()
+
+    def deploy_ocs(self):
+        """
+        Deployment of ODF Managed Service addon on Fusion aaS.
+        """
+        ceph_cluster = ocp.OCP(kind="CephCluster", namespace=self.namespace)
+        try:
+            ceph_cluster.get().get("items")[0]
+            logger.warning("OCS cluster already exists")
+            return
+        except (IndexError, CommandFailed):
+            logger.info("Running OCS basic installation")
+        create_fusion_monitoring_resources()
+        deploy_odf()
+        pod = ocp.OCP(kind=constants.POD, namespace=self.namespace)
+
+        # Check for Ceph pods
+        assert pod.wait_for_resource(
+            condition="Running",
+            selector=constants.MON_APP_LABEL,
+            resource_count=3,
+            timeout=1200,
+        )
+        assert pod.wait_for_resource(
+            condition="Running", selector=constants.MGR_APP_LABEL, timeout=600
+        )
+        assert pod.wait_for_resource(
+            condition="Running",
+            selector=constants.OSD_APP_LABEL,
+            resource_count=3,
+            timeout=600,
+        )
+
+        # Verify health of ceph cluster
+        ceph_health_check(namespace=self.namespace, tries=60, delay=10)
```

Comment:
This section should be part of ocs_ci.ocs.resources.storage_cluster.ocs_install_verification

Reply:
Ok, it will be removed.

Reply:
@fbalak what I meant is that lines 119-136 (the pods status validation + ceph health check) should be placed under ocs_ci.ocs.resources.storage_cluster.ocs_install_verification.

I see you removed ceph_health_check completely. This was not my intention